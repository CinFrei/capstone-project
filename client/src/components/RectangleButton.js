import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

/** @component */
export default RectangleButton

RectangleButton.propTypes = {
  buttonName: PropTypes.string,
  onClick: PropTypes.func,
}

function RectangleButton({
  buttonName,
  gridColumn,
  gridPosition,
  gridRow,
  onClick,
}) {
  return (
    <ButtonStyled
      gridColumn={gridColumn}
      gridPosition={gridPosition}
      gridRow={gridRow}
      onClick={onClick}
    >
      {buttonName}
    </ButtonStyled>
  )
}

const ButtonStyled = styled.button`
  background: var(--accent-color);
  border-radius: 3px;
  border: none;
  color: var(--primary-anthrazit);
  font-family: 'Kaushan Script', cursive;
  font-size: 1.2em;
  grid-column: ${(props) => props.gridColumn};
  grid-row: ${(props) => props.gridRow};
  justify-self: ${(props) => props.gridPosition};
  width: 100%;
`
