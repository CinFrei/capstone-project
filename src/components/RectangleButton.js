import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

/** @component */
export default RectangleButton

RectangleButton.propTypes = {
  buttonName: PropTypes.string,
}

function RectangleButton({ onClick, buttonName }) {
  return <ButtonStyled onClick={onClick}>{buttonName}</ButtonStyled>
}

const ButtonStyled = styled.button`
  width: 100%;
  margin-top: 20px;
  padding: 10px 20px;
  background: var(--light-rose);
  color: var(--primary-antrazit);
  font-size: 1.2em;
  font-family: 'Kaushan Script', cursive;
  border: none;
  border-radius: 3px;
`
