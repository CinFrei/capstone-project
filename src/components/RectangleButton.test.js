import RectangleButton from './RectangleButton'
import { render } from '@testing-library/react'

describe('RectangleButton', () => {
  it('renders correctly', () => {
    const { container } = render(<RectangleButton />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
