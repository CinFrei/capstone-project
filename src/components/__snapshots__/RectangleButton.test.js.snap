// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RectangleButton renders correctly 1`] = `
.c0 {
  background: var(--light-rose);
  border-radius: 3px;
  border: none;
  color: var(--primary-antrazit);
  font-family: 'Kaushan Script',cursive;
  font-size: 1.2em;
  width: 100%;
}

<button
  class="c0"
/>
`;
